src/app/results/components/table.component.ts

```diff
@@ -34,15 +35,80 @@ export class ResultsTableComponent extends AbstractTableComponent<ResultRaw, Res
+  private readonly userService = inject(UserService);
 
   downloadResult$ = new Subject<ArmonikData<ResultRaw>>();
   downloadResultSubscription = this.downloadResult$.subscribe(data => this.onDownload(data.raw.resultId));
 
+  deleteResultData$ = new Subject<ArmonikData<ResultRaw>>();
+  deleteResultDataSubscription = this.deleteResultData$.subscribe(data => this.onDeleteResultData(data.raw.resultId));
+
+  getServiceConfiguration$ = new Subject<ArmonikData<ResultRaw>>();
+  getServiceConfigurationSubscription = this.getServiceConfiguration$.subscribe(data => this.onGetServiceConfiguration(data.raw.resultId));
+
+  uploadResultData$ = new Subject<ArmonikData<ResultRaw>>();
+  uploadResultDataSubscription = this.uploadResultData$.subscribe(data => this.onUploadResultData(data.raw.resultId));
+
+  getResult$ = new Subject<ArmonikData<ResultRaw>>();
+  getResultSubscription = this.getResult$.subscribe(data => this.onGetResult(data.raw.resultId));
+
+  importResultsData$ = new Subject<ArmonikData<ResultRaw>>();
+  importResultsDataSubscription = this.importResultsData$.subscribe(data => this.onImportResultsData(data.raw.resultId));
+
   actions: ActionTable<ResultRaw>[] = [
     {
       label: 'Download result data',
       icon: 'download',
       action$: this.downloadResult$,
+      condition: () => {
+        const permissions = this.userService.user?.permissions ?? [];
+        return permissions.includes('Results:DownloadResultData');
+      },
+    },
+    {
+      label: 'Delete result data',
+      icon: 'delete',
+      action$: this.deleteResultData$,
+      condition: () => {
+        const permissions = this.userService.user?.permissions ?? [];
+        return permissions.includes('Results:DeleteResultsData');
+      },
+    },
+    {
+      label: 'Get service configuration',
+      icon: 'settings',
+      action$: this.getServiceConfiguration$,
+      condition: () => {
+        const permissions = this.userService.user?.permissions ?? [];
+        return permissions.includes('Results:GetServiceConfiguration');
+      },
+    },
+    {
+      label: 'Upload result data',
+      icon: 'upload',
+      action$: this.uploadResultData$,
+      condition: () => {
+        const permissions = this.userService.user?.permissions ?? [];
+        return permissions.includes('Results:UploadResultData');
+      },
+    },
+    {
+      label: 'Get result',
+      icon: 'visibility',
+      action$: this.getResult$,
+      condition: () => {
+        const permissions = this.userService.user?.permissions ?? [];
+        return permissions.includes('Results:GetResult');
+      },
+    },
+    {
+      label: 'Import results data',
+      icon: 'file_download',
+      action$: this.importResultsData$,
+      condition: () => {
+        const permissions = this.userService.user?.permissions ?? [];
+        return permissions.includes('Results:ImportResultsData');
+      },
     },
```

Comment:
No need for all of this. We will add them later if required and if it is implemented if the API.